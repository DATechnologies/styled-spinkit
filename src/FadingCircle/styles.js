// @Flow
import styled, { keyframes } from 'styled-components'
import { propSize, size, propMargin, propBgColor } from '../util'

const opacity = keyframes`
  0%,
  39%,
  100% {
    opacity: 0;
  }

  40% {
    opacity: 1;
  }
`

export const Child = styled.div`
  ${size('100%')}
  position: absolute;
  left: 0;
  top: 0;
  transform: rotate(${p => p.rotate}deg);
  &::before {
    content: '';
    display: block;
    margin: 0 auto;
    ${size('15%')}
    border-radius: 100%;
    animation: ${opacity} 1.2s infinite ease-in-out both -${p => p.delay}s;
  }
`

export const StyledForldingCircle = styled.div`
  ${propSize}
  ${propMargin}
  position: relative;

  > ${Child} {
    &::before {
      ${propBgColor}
    }
  }
`
