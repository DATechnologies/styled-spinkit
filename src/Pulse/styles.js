// @flow
import styled, { keyframes } from 'styled-components'
import { propSize, propBgColor, propMargin } from '../util'

const scaleOut = keyframes`
  0% {
    transform: scale(0);
  }
  100% {
    opacity: 0;
    transform: scale(1);
  }
`

const StyledPulse = styled.div`
  ${propSize};
  ${propBgColor};
  ${propMargin};
  border-radius: 100%;
  animation: ${scaleOut} 1s ease-in-out infinite;
`

export default StyledPulse
