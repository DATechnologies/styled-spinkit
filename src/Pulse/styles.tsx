import styled, { keyframes } from 'styled-components'
import {
  propSize,
  propBgColor,
  propMargin,
  SizeProps,
  BgColorProps,
} from '../util'

const scaleOut = keyframes`
  0% {
    transform: scale(0);
  }
  100% {
    opacity: 0;
    transform: scale(1);
  }
`

type Props = SizeProps & BgColorProps

const StyledPulse = styled.div<Props>`
  ${propSize};
  ${propBgColor};
  ${propMargin};
  border-radius: 100%;
  animation: ${scaleOut} 1s ease-in-out infinite;
`

export default StyledPulse
