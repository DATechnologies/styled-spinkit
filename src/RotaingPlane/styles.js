// @flow

import styled, { keyframes } from 'styled-components'
import { propSize, propBgColor, propMargin } from '../util'

const spin = keyframes`
  0% {
    transform: perspective(120px) rotateX(0deg) rotateY(0deg);
  }
  50% {
    transform: perspective(120px) rotateX(-180.1deg) rotateY(0deg);
  }
  100% {
    transform: perspective(120px) rotateX(-180deg) rotateY(-179.9deg);
  }
`

const StyledRotaingPlain = styled.div`
  ${propSize};
  ${propBgColor};
  ${propMargin};
  animation: ${spin} ${p => p.speed}s infinite ease-in-out;
`

export default StyledRotaingPlain
