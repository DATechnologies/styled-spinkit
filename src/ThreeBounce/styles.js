// @flow
import styled, { keyframes } from 'styled-components'
import { sizePx, propMargin, propBgColor } from '../util'

const bounce = keyframes`
  0%,
  80%,
  100% {
    transform: scale(0);
  }
  40% {
    transform: scale(1);
  }
`

export const Child = styled.div`
  border-radius: 100%;
  display: inline-block;
  animation: ${bounce} 1.4s ease-in-out ${p => p.delay}s infinite both;
`

export const StyledThreeBounce = styled.div`
  width: ${p => p.size}px;
  ${propMargin}
  text-align: center;
  border-radius: 100%;
  > ${Child} {
    ${p => sizePx(p.size / 4)}
    ${propBgColor}
  }
`
