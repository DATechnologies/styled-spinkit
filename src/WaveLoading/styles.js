// @flow
import styled, { keyframes } from 'styled-components'
import { size, propMargin, propBgColor, propDelay } from '../util'

const anim = keyframes`
  0%,
  40%,
  100% {
    transform: scaleY(0.4);
  }
  20% {
    transform: scaleY(1);
  }
`

export const Rect = styled.div`
  ${size('7px', '100%')}
  margin: 0 3px 0 0;
  display: inline-block;
  ${propDelay}
`

export const StyledWave = styled.div`
  ${propMargin}
  ${p => size(`${p.size * 1.25}px`, `${p.size}px`)}
  text-align: center;
  font-size: 10px;

  > ${Rect} {
    ${propBgColor}
    animation-name: ${anim};
    animation-duration: ${p => p.speed}s;
    animation-timing-function: ease-in-out;
    animation-iteration-count: infinite;
  }
`
