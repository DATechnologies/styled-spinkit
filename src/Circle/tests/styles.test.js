// @flow
import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import 'jest-styled-components'
import { Child, StyledCircle } from '../styles'

test('render Child', () => {
  const wrapper = shallow(<Child />)
  expect(toJson(wrapper)).toMatchSnapshot()
})

test('render', () => {
  const wrapper = shallow(<StyledCircle size={40} color={'#333'} />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
