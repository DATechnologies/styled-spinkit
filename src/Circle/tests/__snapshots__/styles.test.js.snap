// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
.c0 {
  margin: 40px auto;
  width: 40px;
  height: 40px;
  position: relative;
}

.c0 > .c1::before {
  background-color: #333;
}

<div
  className="c0"
  color="#333"
  size={40}
/>
`;

exports[`render Child 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: rotate(deg);
  -ms-transform: rotate(deg);
  transform: rotate(deg);
}

.c0::before {
  content: '';
  display: block;
  margin: 0 auto;
  width: 15%;
  height: 15%;
  border-radius: 100%;
  -webkit-animation: ctIBbK 1.2s ease-in-out s infinite both;
  animation: ctIBbK 1.2s ease-in-out s infinite both;
}

<div
  className="c0"
/>
`;
